stan/math/opencl/rev/round.hpp

```diff
@@ -20,10 +20,7 @@ namespace math {
 template <typename T,
           require_all_kernel_expressions_and_none_scalar_t<T>* = nullptr>
 inline var_value<matrix_cl<double>> round(const var_value<T>& A) {
-  return make_callback_var(
-      round(A.val()), [A](vari_value<matrix_cl<double>>& res) mutable {
-        A.adj() = select(isnan(A.val()), NOT_A_NUMBER, A.adj());
-      });
+  return var_value<matrix_cl<double>>(round(A.val()));
```

Comment:
I think you can delete rev overload here and make prim overload handle this (returning prim matrix_cl).

Reply:
Sorry I'm not sure I'm following what you mean here

Reply:
This is a rev overload in which nothing is done in reverse pass, making it essentially the same as prim overload. So we do not need two overloads. `var_value<matrix_cl<double>>` argument can be handled in prim overload.

Reply:
I think the existing behavior is closer to, if autodiff input, then autodiff output. That is what we are doing CPU side too, so I think this is correct as-is.

Where to look for the opencl prim round implementation?

Reply:
Prim round is implemented in kernel generator in `stan/math/opencl/kernel_generator/elt_function_cl.hpp`. Right than it is not that easy to use prim implementation for rev types.